dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -135,6 +136,23 @@ def set_course_from_program
       self.course = PROGRAMS.key(program)
     end
 
+    def set_status_from_admin_approval
+      # Do not modify status is application is not incomplete.
+      return if status == 'incomplete'
+
+      # Do not modify status if admin approval status has not changed.
+      # Since principal_approval_not_required is a serialized attribute, we cannot use the Dirty
+      # API easily –– instead, use the Dirty API to look at the properties attribute.
+      return if properties_change.include?(nil)
+      return unless properties_change.map(&:keys)&.flatten&.include?('principal_approval_not_required')
```

Comment:
This is very janky, but the only way I found to reliably see if the `principal_approval_not_required` field has changed. If you know of other approaches, I'm all ears!

Reply:
This makes sense to me. Big downside of using the serialized attributes in this model :(

Reply:
Amen to that